### ISSUE TITLE
CamelCased names are Capitalized in enums and types

### ISSUE THREAD

**Issue by olbat** [crystal-lang/crystal_lib#42]
Hello,

I'm trying to port a C/C++ lib that defines enums and types which names is written a CamelCased way (it does not really respects C naming convention).

My problem here: when crystal_lib generates my code, every CamelCased names are Capitalized.

Here is an example:
```bash
$ cat /tmp/test.h 
typedef enum TSomeEnum
{
	TSomeValue,
	TSomeOtherValue,
} TSomeEnum;
enum TSomeEnum test();

$ cat test.cr 
@[Include("/tmp/test.h", prefix: "test", remove_prefix: false)]
lib LibTest
end

$ crystal src/main.cr -- test.cr
lib LibTest
  fun test : Tsomeenum
  enum Tsomeenum
    Tsomevalue = 0
    Tsomeothervalue = 1
  end
end
```

As you can see, the `TSomeEnum` type is transformed to `Tsomeenum` and `TSomeValue` is transformed to `Tsomevalue`.
This can be quite confusing when using the crystal lib that was generated this way since constants and types names does not match the C ones (and it's doc).

Is this a desired behavior ? If yes, is there any (easy) way I can make the names being generated in CamelCase ?

**Comment by mverzilli** [comment]
I don't think this is desired, so it's a bug. The problem seems to be here: https://github.com/crystal-lang/crystal_lib/blob/master/src/crystal_lib/type_mapper.cr#L244

And judging by the look of the specs: https://github.com/crystal-lang/crystal_lib/blob/master/spec/parser_spec.cr#L181, it seems this case was simply not accounted for.

Do you want to give it a try? It should be a matter of adding a test to parser_spec and then enhancing the `crystal_type_name` function :).

**Comment by olbat** [comment]
> Do you want to give it a try? It should be a matter of adding a test to parser_spec and then enhancing the crystal_type_name function :).

No problem, I'll try to propose a fix today (sry for the late answer).

**Comment by olbat** [comment]
Here it is #44 .

(I think there was [something wrong in the tests](https://github.com/crystal-lang/crystal_lib/blob/master/spec/lib_body_transformer_spec.cr#L112-L128) and proposed a fix in my first commit, just let me know if it was a mistake)

**Comment by mverzilli** [comment]
Thanks! I'll take a look soon.

**Comment by mverzilli** [comment]
Fixed through #44
